packages/anoncreds/src/AnonCredsApi.ts

```diff
@@ -30,24 +39,71 @@ export class AnonCredsApi {
   private anonCredsRegistryService: AnonCredsRegistryService
   private anonCredsSchemaRepository: AnonCredsSchemaRepository
   private anonCredsCredentialDefinitionRepository: AnonCredsCredentialDefinitionRepository
+  private anonCredsCredentialDefinitionPrivateRepository: AnonCredsCredentialDefinitionPrivateRepository
+  private anonCredsKeyCorrectnessProofRepository: AnonCredsKeyCorrectnessProofRepository
+  private anonCredsMasterSecretRepository: AnonCredsMasterSecretRepository
 
   // TODO: how do we inject the anoncreds services?
   private anonCredsIssuerService: AnonCredsIssuerService
+  private anonCredsHolderService: AnonCredsHolderService
 
   public constructor(
     agentContext: AgentContext,
     anonCredsRegistryService: AnonCredsRegistryService,
     config: AnonCredsModuleConfig,
     anonCredsIssuerService: AnonCredsIssuerService,
+    anonCredsHolderService: AnonCredsHolderService,
     anonCredsSchemaRepository: AnonCredsSchemaRepository,
-    anonCredsCredentialDefinitionRepository: AnonCredsCredentialDefinitionRepository
+    anonCredsCredentialDefinitionRepository: AnonCredsCredentialDefinitionRepository,
+    anonCredsCredentialDefinitionPrivateRepository: AnonCredsCredentialDefinitionPrivateRepository,
+    anonCredsKeyCorrectnessProofRepository: AnonCredsKeyCorrectnessProofRepository,
+    anonCredsMasterSecretRepository: AnonCredsMasterSecretRepository
   ) {
     this.agentContext = agentContext
     this.anonCredsRegistryService = anonCredsRegistryService
     this.config = config
     this.anonCredsIssuerService = anonCredsIssuerService
+    this.anonCredsHolderService = anonCredsHolderService
     this.anonCredsSchemaRepository = anonCredsSchemaRepository
     this.anonCredsCredentialDefinitionRepository = anonCredsCredentialDefinitionRepository
+    this.anonCredsCredentialDefinitionPrivateRepository = anonCredsCredentialDefinitionPrivateRepository
+    this.anonCredsKeyCorrectnessProofRepository = anonCredsKeyCorrectnessProofRepository
+    this.anonCredsMasterSecretRepository = anonCredsMasterSecretRepository
+  }
+
+  /**
+   * Create a Master Secret, optionally indicating its ID and if it will be the default one
+   * If there is no other Master Secret, this will be the default one.
+   *
+   */
+  public async createMasterSecret(options?: AnonCredsCreateMasterSecretOptions) {
```

Comment:
masterSecret -> linkSecret

```suggestion
  public async createLinkSecret(options?: AnonCredsCreateMasterSecretOptions) {
```